compaction/size_tiered_compaction_strategy.cc

```diff
@@ -54,8 +56,10 @@ size_tiered_compaction_strategy::create_sstable_and_length_pairs(const std::vect
     return sstable_length_pairs;
```

Comment:
It doesn't even solve any exception guarantees, since we can throw here.

Isn't it better to convert to a vector at the source, than to duplicate the entire function?

Reply:
But that would add an allocation.  I'm not sure what would be better, but I'll go with your more minimalistic approach and drop this patch from next version.

Reply:
`_all` copied into a vector in v6